src/main.cpp

```diff
@@ -253,6 +315,45 @@ void setup()
   digitalWrite(MODEM_RST, HIGH);
   digitalWrite(MODEM_POWER_ON, HIGH);
 
+  // Set GSM module baud rate and UART pins
+  Serial.println("[  INIT  ] Initializing serial interface to modem...");
+  SerialAT.begin(115200, SERIAL_8N1, MODEM_RX, MODEM_TX);
+  delay(3000);
+  initStage++;
+
+  // Restart takes quite some time
+  // use modem.init() if you don't need the complete restart
+  Serial.println("[  INIT  ] Initializing modem...");
+  modem.restart();
+  
+  String name = modem.getModemName();
+  Serial.print("[  INIT  ] Modem Name: ");
+  Serial.print(name);
+  String modemInfo = modem.getModemInfo();
+  Serial.print(", Modem Info: ");
+  Serial.println(modemInfo);
+  initStage++;
+
+  // Unlock your SIM card with a PIN if needed
+  // if (strlen(simPIN) && modem.getSimStatus() != 3 ) {
+  //   Serial.println("Unlocking SIM...");
+  //   modem.simUnlock(simPIN);
+  // }
+
+  Serial.print("[  INIT  ] Waiting for network...");
+  if (!modem.waitForNetwork(240000L)) {
```

Comment:
das müsste man in der loop() machen, damit der bootprozess nicht durch schlechten Empfang usw. blockiert wird.

Reply:
Ich habe jetzt mal in der 30s Loop immer auf isNetwork Connected geprüft. Das geht aber atürlich beim ersten Mal schief und damit ist auch für den ersten Bildschirmaufbau kein Datum (wenn wir denn eins kriegen würden) da. Wenn ich aber jetzt die 300s Task verzögere habe ich gegenüber der Umsetzung in der init() auch nicht wirklich was gewonnen, oder?

Reply:
Ich habs jetzt nochmal anders gemacht ...